ragna/ui/api_wrapper.py

```diff
@@ -0,0 +1,108 @@
+import os
+import re
+from datetime import datetime
+
+import emoji
+
+import httpx
+
+
+# The goal is this class is to provide ready-to-use functions to interact with the API
+class ApiWrapper:
+    def __init__(self, api_url):
+        # FIXME: this should be an async client
+        self.client = httpx.Client(base_url=api_url)
+        self.client.get("/").raise_for_status()
+        # FIXME: the token should come from a cookie that is set after UI login
+        user = "User"
+        token = (
+            self.client.post(
+                "/token",
+                data={
+                    "username": user,
+                    "password": os.environ.get(
+                        "AI_PROXY_DEMO_AUTHENTICATION_PASSWORD", user
+                    ),
+                },
+            )
+            .raise_for_status()
+            .json()
+        )
+        # FIXME: remove this as it should come from a cookie on the JS side as well
+        self.token = token
+        self.client.headers["Authorization"] = f"Bearer {token}"
+
+    def get_chats(self):
+        json_data = self.client.get("/chats").raise_for_status().json()
+
+        for chat in json_data:
+            for msg in chat["messages"]:
+                # convert timestamps to datetime
+                msg["timestamp"] = datetime.strptime(
+                    msg["timestamp"], "%Y-%m-%dT%H:%M:%S.%f"
+                )
+
+                msg["content"] = self.replace_emoji_shortcodes_with_emoji(
+                    msg["content"]
+                )
```

Comment:
This is why the emoji rendering works after page reload. We should apply this function to every message that gets added to the chat widget.